doc/devel/coding_style_guideline.rst

```diff
@@ -0,0 +1,93 @@
+.. _coding_style_guideline:
+
+===========================
+DIPY Coding Style Guideline
+===========================
+
+The main principles behind dipy_ development are:
+* **Robustness**: the results of a piece of code must be verified
+systematically, and hence stability and robustness of the code must be ensured,
+reducing code redundancies.
+* **Readability**: the code is written and read by humans, and it is read much
+more frequently than it is written.
+* **Consistency**: following these guidelines will ease reading the code, and
+will make it less error-prone.
+* **Documentation**: document the code. Documentation is essential as it is one
+of the key points for the adoption of DIPY as the toolkit of choice in
+diffusion by the scientific community. Documenting helps clarifying certain
+choices, helps avoiding obscure places, and is a way to allow other members
+*decode* it with less effort.
+* **Language**: the code must be written in English. Norms and spelling should
+be abided by.
+
+
+-----------
+DIPY core
+-----------
+
+DIPY has adopted the standard Python `PEP8
+<https://www.python.org/dev/peps/pep-0008/>_` style to ensure the
+readability and consistency across the toolkit. The PEP8 style is enforced
+automatically when requesting to push to DIPY. However, as a contributor to
+DIPY, you should try to ensure that your code, including its comments
+
+
```

Comment:
and you can cite the Numpy detailed example [here](https://github.com/numpy/datetime/blob/master/doc/EXAMPLE_DOCSTRING.txt)